File: _COMMUNICATION/team_31/team_31_staging/scan-trading-accounts-tables.js
/**
 * Scanner for Trading Accounts Page Tables
 * Scans the live legacy server at port 8090 to extract all table structures
 */

(async function scanTradingAccountsTables() {
  const BASE_URL = 'http://127.0.0.1:8090';
  const LOGIN_URL = `${BASE_URL}/login`;
  const TRADING_ACCOUNTS_URL = `${BASE_URL}/trading_accounts`;
  
  console.log('🔍 Starting Trading Accounts Tables Scanner...\n');
  
  try {
    // Step 1: Login
    console.log('📝 Step 1: Logging in...');
    const loginResponse = await fetch(LOGIN_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
      body: 'username=admin&password=admin123',
      credentials: 'include'
    });
    
    if (!loginResponse.ok) {
      throw new Error(`Login failed: ${loginResponse.status}`);
    }
    
    const cookies = loginResponse.headers.get('set-cookie');
    console.log('✅ Login successful\n');
    
    // Step 2: Fetch Trading Accounts Page
    console.log('📄 Step 2: Fetching Trading Accounts page...');
    const pageResponse = await fetch(TRADING_ACCOUNTS_URL, {
      credentials: 'include',
      headers: {
        'Cookie': cookies || ''
      }
    });
    
    if (!pageResponse.ok) {
      throw new Error(`Failed to fetch page: ${pageResponse.status}`);
    }
    
    const htmlContent = await pageResponse.text();
    console.log('✅ Page fetched\n');
    
    // Step 3: Parse HTML and extract table structures
    console.log('🔎 Step 3: Analyzing page structure...');
    
    // Extract all tables
    const tableRegex = /<table[^>]*>[\s\S]*?<\/table>/gi;
    const tables = htmlContent.match(tableRegex) || [];
    
    console.log(`📊 Found ${tables.length} table(s) in HTML\n`);
    
    // Extract table headers and structure
    const tableStructures = [];
    
    tables.forEach((tableHtml, index) => {
      console.log(`\n📋 Table ${index + 1}:`);
      
      // Extract table ID or class
      const idMatch = tableHtml.match(/id=["']([^"']+)["']/i);
      const classMatch = tableHtml.match(/class=["']([^"']+)["']/i);
      const dataTableType = tableHtml.match(/data-table-type=["']([^"']+)["']/i);
      
      const tableInfo = {
        index: index + 1,
        id: idMatch ? idMatch[1] : null,
        className: classMatch ? classMatch[1] : null,
        dataTableType: dataTableType ? dataTableType[1] : null,
        columns: []
      };
      
      // Extract headers
      const theadMatch = tableHtml.match(/<thead[^>]*>([\s\S]*?)<\/thead>/i);
      if (theadMatch) {
        const thMatches = theadMatch[1].match(/<th[^>]*>([\s\S]*?)<\/th>/gi) || [];
        thMatches.forEach((th, colIndex) => {
          const textContent = th.replace(/<[^>]+>/g, '').trim();
          const sortable = th.includes('sortable') || th.includes('data-sort');
          const sortKey = th.match(/data-sort-key=["']([^"']+)["']/i)?.[1] || null;
          
          tableInfo.columns.push({
            index: colIndex,
            header: textContent,
            sortable: sortable,
            sortKey: sortKey
          });
        });
      }
      
      // Extract sample rows (first 3 rows)
      const tbodyMatch = tableHtml.match(/<tbody[^>]*>([\s\S]*?)<\/tbody>/i);
      if (tbodyMatch) {
        const rowMatches = tbodyMatch[1].match(/<tr[^>]*>([\s\S]*?)<\/tr>/gi) || [];
        tableInfo.sampleRows = rowMatches.slice(0, 3).map(row => {
          const cellMatches = row.match(/<t[dh][^>]*>([\s\S]*?)<\/t[dh]>/gi) || [];
          return cellMatches.map(cell => {
            return cell.replace(/<[^>]+>/g, '').trim();
          });
        });
      }
      
      tableStructures.push(tableInfo);
      
      console.log(`  ID: ${tableInfo.id || 'N/A'}`);
      console.log(`  Class: ${tableInfo.className || 'N/A'}`);
      console.log(`  Data Table Type: ${tableInfo.dataTableType || 'N/A'}`);
      console.log(`  Columns (${tableInfo.columns.length}):`);
      tableInfo.columns.forEach(col => {
        console.log(`    ${col.index}: ${col.header}${col.sortable ? ' [SORTABLE]' : ''}${col.sortKey ? ` (key: ${col.sortKey})` : ''}`);
      });
      
      if (tableInfo.sampleRows && tableInfo.sampleRows.length > 0) {
        console.log(`  Sample Rows: ${tableInfo.sampleRows.length}`);
        tableInfo.sampleRows.forEach((row, rowIndex) => {
          console.log(`    Row ${rowIndex + 1}: ${row.join(' | ')}`);
        });
      }
    });
    
    // Step 4: Look for dynamically generated tables (check for table containers)
    console.log('\n\n🔎 Step 4: Looking for table containers and dynamic structures...');
    
    // Look for common table container patterns
    const containerPatterns = [
      /<div[^>]*class=["'][^"']*table[^"']*["'][^>]*>/gi,
      /<div[^>]*id=["'][^"']*table[^"']*["'][^>]*>/gi,
      /<section[^>]*class=["'][^"']*table[^"']*["'][^>]*>/gi,
    ];
    
    const containers = [];
    containerPatterns.forEach(pattern => {
      const matches = htmlContent.match(pattern) || [];
      matches.forEach(match => {
        const idMatch = match.match(/id=["']([^"']+)["']/i);
        const classMatch = match.match(/class=["']([^"']+)["']/i);
        containers.push({
          id: idMatch ? idMatch[1] : null,
          className: classMatch ? classMatch[1] : null,
          html: match
        });
      });
    });
    
    console.log(`📦 Found ${containers.length} potential table containers`);
    
    // Step 5: Output results
    console.log('\n\n📊 SCAN RESULTS SUMMARY:');
    console.log('='.repeat(60));
    console.log(JSON.stringify({
      timestamp: new Date().toISOString(),
      url: TRADING_ACCOUNTS_URL,
      tablesFound: tables.length,
      tableStructures: tableStructures,
      containersFound: containers.length
    }, null, 2));
    
    // Save to file
    const fs = require('fs');
    const path = require('path');
    const outputPath = path.join(__dirname, 'trading-accounts-tables-scan.json');
    fs.writeFileSync(outputPath, JSON.stringify({
      timestamp: new Date().toISOString(),
      url: TRADING_ACCOUNTS_URL,
      tablesFound: tables.length,
      tableStructures: tableStructures,
      containersFound: containers.length
    }, null, 2));
    
    console.log(`\n✅ Results saved to: ${outputPath}`);
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    console.error(error.stack);
  }
})();

File: _COMMUNICATION/team_31/team_31_staging/scan-trading-accounts-tables.py
#!/usr/bin/env python3
"""
Scanner for Trading Accounts Page Tables
Scans the live legacy server at port 8090 to extract all table structures
"""

import requests
from bs4 import BeautifulSoup
import json
from datetime import datetime

BASE_URL = 'http://127.0.0.1:8090'
LOGIN_URL = f'{BASE_URL}/login'
TRADING_ACCOUNTS_URL = f'{BASE_URL}/trading_accounts'

def scan_trading_accounts_tables():
    print('🔍 Starting Trading Accounts Tables Scanner...\n')
    
    session = requests.Session()
    
    try:
        # Step 1: Login
        print('📝 Step 1: Logging in...')
        login_data = {
            'username': 'admin',
            'password': 'admin123'
        }
        login_response = session.post(LOGIN_URL, data=login_data)
        
        if login_response.status_code != 200:
            raise Exception(f'Login failed: {login_response.status_code}')
        
        print('✅ Login successful\n')
        
        # Step 2: Fetch Trading Accounts Page
        print('📄 Step 2: Fetching Trading Accounts page...')
        page_response = session.get(TRADING_ACCOUNTS_URL)
        
        if page_response.status_code != 200:
            raise Exception(f'Failed to fetch page: {page_response.status_code}')
        
        html_content = page_response.text
        print('✅ Page fetched\n')
        
        # Step 3: Parse HTML
        print('🔎 Step 3: Parsing HTML...')
        soup = BeautifulSoup(html_content, 'html.parser')
        
        # Find all tables
        tables = soup.find_all('table')
        print(f'📊 Found {len(tables)} table(s) in HTML\n')
        
        report = {
            'timestamp': datetime.now().isoformat(),
            'url': TRADING_ACCOUNTS_URL,
            'tablesFound': len(tables),
            'tables': [],
            'containers': []
        }
        
        # Extract table structures
        for table_index, table in enumerate(tables):
            print(f'\n📋 Table {table_index + 1}:')
            
            table_info = {
                'index': table_index + 1,
                'id': table.get('id'),
                'className': ' '.join(table.get('class', [])),
                'dataTableType': table.get('data-table-type'),
                'columns': [],
                'sampleRows': []
            }
            
            # Find container
            container = table.find_parent(['div', 'section'], class_=lambda x: x and ('content-section' in x or 'section' in x.lower()))
            if container:
                container_title = container.find(['h2', 'h3', 'div'], class_=lambda x: x and ('title' in str(x).lower() or 'header' in str(x).lower()))
                if container_title:
                    table_info['container'] = {
                        'id': container.get('id'),
                        'className': ' '.join(container.get('class', [])),
                        'title': container_title.get_text(strip=True)
                    }
            
            # Extract headers
            thead = table.find('thead')
            if thead:
                headers = thead.find_all('th')
                for col_index, th in enumerate(headers):
                    header_text = th.get_text(strip=True)
                    is_sortable = 'sortable' in ' '.join(th.get('class', [])).lower() or th.get('data-sort') is not None
                    sort_key = th.get('data-sort-key') or th.get('data-sort')
                    sort_type = th.get('data-sort-type')
                    
                    # Extract column classes
                    col_classes = [cls for cls in th.get('class', []) if cls.startswith('col-')]
                    
                    table_info['columns'].append({
                        'index': col_index,
                        'header': header_text,
                        'sortable': is_sortable,
                        'sortKey': sort_key,
                        'sortType': sort_type,
                        'columnClasses': col_classes
                    })
            
            # Extract sample rows (first 5 rows)
            tbody = table.find('tbody')
            if tbody:
                rows = tbody.find_all('tr', limit=5)
                for row_index, row in enumerate(rows):
                    cells = row.find_all('td')
                    row_data = {
                        'rowIndex': row_index,
                        'cells': []
                    }
                    
                    for cell_index, cell in enumerate(cells):
                        cell_text = cell.get_text(strip=True)
                        cell_classes = cell.get('class', [])
                        col_classes = [cls for cls in cell_classes if cls.startswith('col-')]
                        
                        row_data['cells'].append({
                            'index': cell_index,
                            'text': cell_text,
                            'classes': cell_classes,
                            'columnClasses': col_classes
                        })
                    
                    table_info['sampleRows'].append(row_data)
            
            report['tables'].append(table_info)
            
            print(f'  ID: {table_info["id"] or "N/A"}')
            print(f'  Container: {table_info.get("container", {}).get("title", "N/A")}')
            print(f'  Columns: {len(table_info["columns"])}')
            for col in table_info['columns']:
                sortable_str = ' [SORTABLE]' if col['sortable'] else ''
                col_classes_str = f' ({", ".join(col["columnClasses"])})' if col['columnClasses'] else ''
                print(f'    {col["index"]}: "{col["header"]}"{sortable_str}{col_classes_str}')
            print(f'  Sample Rows: {len(table_info["sampleRows"])}')
        
        # Find containers/sections
        containers = soup.find_all(['div', 'section'], class_=lambda x: x and ('content-section' in str(x) or 'section' in str(x).lower()))
        for container_index, container in enumerate(containers):
            title_elem = container.find(['h2', 'h3', 'div'], class_=lambda x: x and ('title' in str(x).lower() or 'header' in str(x).lower()))
            title = title_elem.get_text(strip=True) if title_elem else None
            tables_in_container = container.find_all('table')
            
            if title or len(tables_in_container) > 0:
                report['containers'].append({
                    'index': container_index + 1,
                    'id': container.get('id'),
                    'className': ' '.join(container.get('class', [])),
                    'title': title,
                    'tableCount': len(tables_in_container)
                })
        
        print(f'\n\n📦 Found {len(report["containers"])} containers/sections')
        for container in report['containers']:
            print(f'  {container["index"]}: {container["title"] or container["id"] or "Untitled"} ({container["tableCount"]} tables)')
        
        # Save report
        output_file = 'trading-accounts-tables-scan.json'
        with open(output_file, 'w', encoding='utf-8') as f:
            json.dump(report, f, ensure_ascii=False, indent=2)
        
        print(f'\n✅ Results saved to: {output_file}')
        print('\n📊 FULL REPORT:')
        print('=' * 80)
        print(json.dumps(report, ensure_ascii=False, indent=2))
        print('=' * 80)
        
        return report
        
    except Exception as error:
        print(f'❌ Error: {error}')
        import traceback
        traceback.print_exc()
        return None

if __name__ == '__main__':
    scan_trading_accounts_tables()
